### PR TITLE
feature-34-home-lambda-api-specs-for-sprint-view

### LINKED ISSUE #34: Home Lambda API specs for sprint-view
Creating lambda API specs for functions to retrieve resourceAllocations, Projects, Phases(Tasks before), and WorkHours

### PR DESCRIPTION
**Pull request by naStipe**
Added schemas and function specs for ResourceAllocations, WorkHours and Phases

### CODE REVIEW

**Review by Hanhz2707 — APPROVED** [review]

**Review by AfoninOleksandr — APPROVED** [review]
Looks good

File: swagger.yaml [hunk 1]
```diff
@@ -1436,21 +1528,18 @@ components:
           description: Unique identifier for the software entry
         name:
           type: string
-          description: Name of the software entry
```
Comment (dalju005-xamk) [inline swagger.yaml:1439]: What are these?

File: swagger.yaml [hunk 2]
```diff
@@ -721,15 +721,15 @@ paths:
                 items:
                   $ref: "#/components/schemas/Error"
 
-  "/severa/{severaGuid}/flextime":
+  "/severa/{severaUserGuid}/flextime":
```
Comment (dalju005-xamk) [inline swagger.yaml]: This needs to be changes to other places also. Where it now severaGuid. In example in keycloak-Api-service. Maybe it on bugfix PR ?

**Review by dalju005-xamk — APPROVED** [review]

File: swagger.yaml (hunk 2 above)
Comment (Jdallos) [reply swagger.yaml]: I thought we discussed the idea of renaming all references to a Guid to Id. I think this is the best options. So in this case it should be severaUserId. Please ensure all other references to Guid are upated

Comment (Jdallos) [reply swagger.yaml]: I won't leave any other comments relating to this change so please check

File: swagger.yaml [hunk 3]
```diff
@@ -751,7 +751,99 @@ paths:
                 items:
                   $ref: "#/components/schemas/Error"
 
-  "/vacationsRequests":
+  "/severa/projects/{severaProjectGuid}/phaseswithhierarchy":
```
Comment (Jdallos) [inline swagger.yaml]: I'm not sure about the endpoint name of phaseswithhierarchy, I'd suggest just phases

File: swagger.yaml [hunk 4]
```diff
@@ -751,7 +751,99 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Error"
+
+  "/severa/users/{severaUserGuid}/workhours":
+    get:
+      operationId: getWorkHoursBySeveraProjectId
```
Comment (Jdallos) [inline swagger.yaml]: Is it the workhours for a project?  I see only the userId in use, how do we define which project? Or is the id and summary incorrectly named?

File: swagger.yaml [hunk 5]
```diff
@@ -751,7 +751,99 @@ paths:
...
+      summary: Get all the work hours for project
+      description: Retrieve all the work hours for a particular user with given ID.
+      tags:
+        - WorkHours
+      parameters:
+        - name: severaUserGuid
+          in: path
+          required: true
+          description: ID of the work hour.
```
Comment (Jdallos) [inline swagger.yaml]: Isn't it the ID of the user?

File: swagger.yaml (hunk 4 above)
Comment (Jdallos) [reply swagger.yaml]: Looking at the workhours schema I assume we can define the project/ phase the workhours refer to. I assume this is done using query params, in which case there query params should be defined here. Are some of them optional?

File: swagger.yaml [hunk 6]
```diff
@@ -751,7 +751,99 @@ paths:
...
+        "default":
+          description: Invalid request was sent to the server.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/Error"
+
+  "/severa/{severaUserGuid}/resourceallocations":
```
Comment (Jdallos) [inline swagger.yaml]: Use camelCase for the endpoint naming e.g. resourceAllocations

**Review by Jdallos — CHANGES_REQUESTED** [review]

File: swagger.yaml (hunk 3 above)
Comment (naStipe) [reply swagger.yaml]: There is no endpoint "phases" for user. this endpoint receives all phases for a particular user. Do you want us to retrieve all phases and then sort the ones that we need?

File: swagger.yaml (hunk 4 above)
Comment (naStipe) [reply swagger.yaml]: Yes. All of them are optional, but we will add the lacking params

File: swagger.yaml (hunk 5 above)
Comment (naStipe) [reply swagger.yaml]: Yes, it is. We will fix the description.

File: swagger.yaml (hunk 6 above)
Comment (naStipe) [reply swagger.yaml]: Does it matter how they are written in severa? These endpoint were "copied" from the docs

File: swagger.yaml (hunk 4 above)
Comment (naStipe) [reply swagger.yaml]: Actually, they are not defined in the params. Project/phase is a property of workhours. So they should be stored in the same entity in the database, not defined manually in params. Let me know if I misunderstood your question.

File: swagger.yaml (hunk 3 above)
Comment (naStipe) [reply swagger.yaml]: All phases of a particular project* 
Not user

File: swagger.yaml [hunk 7]
```diff
@@ -751,7 +751,132 @@ paths:
...
+          schema: 
+            type: string
+            format: date-time
+        - name: eventDate
+          required: false
+          in: query
+          description: Filter work hours by eventDate
+          schema: 
+            type: string
+            format: date
```
Comment (naStipe) [inline swagger.yaml:862]: Not sure if we need to filter by eventDate. Maybe we should delete it for now?

File: swagger.yaml [hunk 8]
```diff
@@ -751,7 +751,125 @@ paths:
...
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Phase"
+        "default":
+          description: Invalid request was sent to the server.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
```
Comment (Jdallos) [inline swagger.yaml]:
```suggestion
```

Comment (Jdallos) [reply swagger.yaml]: I noticed this crept into the spec at some point, we don't need these two lines so they can be removed.

File: swagger.yaml [hunk 9]
```diff
@@ -751,7 +751,125 @@ paths:
...
+                  $ref: "#/components/schemas/ResourceAllocations"
+        "default":
+          description: Invalid request was sent to the server.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
```
Comment (Jdallos) [inline swagger.yaml]:
```suggestion
```

**Review by Jdallos — APPROVED** [review]

**Pull request merged**